### PR TITLE
Filter option for many-to-many objects relations

### PR DESCRIPTION
**Pull request by jremmurd**
# New Feature: Filter for many-to-many objects relations

This pull request provides a new configuration option for `many-to-many objects relations` and `advanced many-to-many object relations`, which enables a simple text-filter in the panel's header. Filtering by text- or numbers-fields and also links are supported. 

image

### CODE REVIEW

**Comment by jremmurd** [comment]
@brusch Thanks for your feedback. I made some adjustments: as you proposed, the filter button is now always visible and toggles the filter input. `Blur `event on an empty filter hides the filter input again.

image

image

File: doc/Development_Documentation/05_Objects/01_Object_Classes/01_Data_Types/09_Relation_Types.md [hunk 1]
```diff
@@ -96,6 +96,9 @@ $relationObject = $relation->getObject();
 DataObject\AbstractObject::setHideUnpublished(true);
 ```
 
+### Many-To-Many Object Relation
+If enabled in the class definition field settings a filter will be displayed. The filter currently supports filtering texts, numbers and links.
```
Comment (fashxp) [inline doc/Development_Documentation/05_Objects/01_Object_Classes/01_Data_Types/09_Relation_Types.md]: this comment in docs needs to be removed again ;-)

Comment (jremmurd) [reply doc/Development_Documentation/05_Objects/01_Object_Classes/01_Data_Types/09_Relation_Types.md]: thanks, done!

**Comment by brusch** [comment]
Thanks a lot!

**Pull request merged**